### ISSUE TITLE
[known issue] kthvalue not available on CUDA

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#2134]
```torch.arange(1, 10).cuda().kthvalue(5)``` will throw

A known issue (https://discuss.pytorch.org/t/missing-kthvalue-on-gpu/1798/2)

**Comment by apaszke** [comment]
A workaround for now would be to use `topk`. It should be very fast.
```python
def kthvalue(tensor, k, dim=None):
    val, idx = tensor.topk(k, dim)
    return val[:1], idx[:1]
```

**Comment by vadimkantorov** [comment]
`kthvalue` returns k'th smallest element, while `topk` returns by default largest elements. So a workaround should be:
```python
 # dim = None wouldn't work because topk requires a dim argument if largest = False is used
def kthvalue(tensor, k, dim=-1):
    val, idx = tensor.topk(k, dim, largest = False)
    return val[-1:], idx[-1:]
```

**Comment by AnishShah** [comment]
Hi, can I work on this?

**Comment by soumith** [comment]
sure thing @AnishShah !

**Comment by vadimkantorov** [comment]
fix worked on in #8406

**Comment by vadimkantorov** [comment]
Fixed in https://github.com/pytorch/pytorch/pull/17544
